zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -361,6 +361,8 @@ private void batchProcessing(final TypedRecord<?> initialCommand) {
 
       final var command = pendingCommands.removeFirst();
 
+      processingResultBuilder.setRequestId(command.getRequestId());
```

Comment:
❌ Setting `requestId` is an exceptional case here and non-trivial. Let's add a comment on top of it and mention the intention that we would like to set the initial `requestId` to followup commands as well inside `ProcessingResultBuilder` implementations and eventually export it for each record.

That way, other developers can reason about the decision and maintain the related code easily in the future.